src/Rules/NoDebugInNamespace.php

```diff
@@ -0,0 +1,59 @@
+<?php declare(strict_types=1);
+
+namespace Hihaho\PhpstanRules\Rules;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\FuncCall;
+use PHPStan\Analyser\Scope;
+use PHPStan\Rules\Rule;
+use PHPStan\Rules\RuleErrorBuilder;
+
+/**
+ * @implements \PHPStan\Rules\Rule<\PhpParser\Node\Expr\FuncCall>
+ */
+class NoDebugInNamespace implements Rule
+{
+    public function getNodeType(): string
+    {
+        return FuncCall::class;
+    }
+
+    public function processNode(Node $node, Scope $scope): array
+    {
+        if (! $node->name instanceof Node\Name) {
+            return [];
+        }
+
+        if ($this->hasDisallowedStatements($node) && str_starts_with($scope->getNamespace(), 'App')) {
```

Comment:
We now check `$this->hasDisallowedStatements($node)` twice when not in the App namespace, perhaps we can do an early return when there aer no disallowed statements

Reply:
Oeh that's interesting. Will try that